apps/web/src/components/forms/user.tsx

```diff
@@ -22,13 +23,17 @@ export default function UserForm({ workspaceId, onSubmit }: UserFormProps) {
   const { register, formState, handleSubmit } = useForm<UserFormValues>({
     mode: 'onChange',
   })
+  const [loading, setLoading] = useState(false)
```

Comment:
`react-hook-form` has an [`isSubmitting`](https://react-hook-form.com/docs/useform/formstate#:~:text=any%20runtime%20error.-,isSubmitting,-boolean) boolean inside `formState`, which is true while the `handleSubmit` promise is resolving.

I quick tested and it does work with `useCallback`, so it will help you save some memory by removing this state.

Reply:
Loved it, you're the best! I'll use it.